File: pattern_matching.r
pattern_matching <- function(pattern, genome) {
  pattern_length <- nchar(pattern)
  for (i in 0:nchar(genome) - pattern_length) {
    if (substr(genome, i + 1, i + pattern_length) == pattern)
      print(i)
  }
}


message("File")
file <- scan("stdin", what="character", nlines=1)
pattern <- scan(file, what="character", nlines=1)
genome <- scan(file, what="character", skip=1, nlines=1)

pattern_matching(pattern, genome)

File: pattern_matching.py
from hamming_distance import hamming_distance

def pattern_matching(pattern, genome, d):
  pattern_length = len(pattern)
  pos = []
  for i in range(0, len(genome) - pattern_length + 1):
    _pattern = genome[i:i + pattern_length]
    if (hamming_distance(pattern, _pattern) <= d):
      pos.append(i)
  return pos


if __name__ == "__main__":
  filename = raw_input("Enter a file name: ")
  f = open(filename, "r")
  pattern = f.readline().rstrip('\r\n')
  genome = f.readline().rstrip('\r\n')
  d = int(f.readline())
  " ".join(map(str, pattern_matching(pattern, genome, d)))
